### ISSUE TITLE
Handle error in into_reader

### ISSUE THREAD

**Issue by asv1** [algesten/ureq#55]
Cannot handle error in such case:
```
use std::io;

fn main() {
    let mut reader = ureq::get("https:://123").call().into_reader();
    let mut out = Vec::new(); // error message will be there
    io::copy(&mut reader, &mut out).unwrap(); // no error

    dbg!(String::from_utf8_lossy(&out));
}
```

Looks like no error propagate from read to io::copy.

**Comment by algesten** [comment]
This needs investigation.

**Comment by jsha** [comment]
I think ureq is working as intended. @asv1's code is missing an is_ok, error, or synthetic_error call:

```
fn main() {
    let mut resp = ureq::get("https:://123").call();
    if let Some(error) = resp.synthetic_error() {
        eprintln!("{}", error);
        return;
    }
    let mut reader = resp.into_reader();
    let mut out = Vec::new(); // error message will be there
    io::copy(&mut reader, &mut out).unwrap(); // no error

    dbg!(String::from_utf8_lossy(&out));
}
```

That said, I did the same thing when I first tried out ureq, and was very surprised and thought it was a bug. ureq's approach of turning errors into synthetic response codes, documented at https://docs.rs/ureq/1.1.2/ureq/struct.Response.html#method.synthetic, is definitely a bit idiosyncratic.

I think the right fix here is to improve the documentation. For instance, the top-level example for using Response, at https://docs.rs/ureq/1.1.2/ureq/struct.Response.html, should include error handling. Similarly, the overall API doc example at https://docs.rs/ureq/1.1.2/ureq/index.html just has a check for `resp.ok()`. More detailed error handling in that example, along with an example of reading the stream, would help avoid confusion.

I'll work on this today.
